src/plugins/cjs-interop.ts

```diff
@@ -1,26 +1,125 @@
+import type {
+  ExportDefaultExpression,
+  ModuleDeclaration,
+  ParseOptions,
+} from '@swc/core'
+import type { Visitor } from '@swc/core/Visitor'
+import fs from 'fs/promises'
+import path from 'path'
+import { PrettyError } from '../errors'
 import { Plugin } from '../plugin'
+import { localRequire } from '../utils'
 
 export const cjsInterop = (): Plugin => {
   return {
     name: 'cjs-interop',
 
     async renderChunk(code, info) {
+      const { entryPoint } = info
       if (
         !this.options.cjsInterop ||
         this.format !== 'cjs' ||
         info.type !== 'chunk' ||
         !/\.(js|cjs)$/.test(info.path) ||
```

Comment:
```suggestion
        !/\.(js|mjs)$/.test(info.path) ||
```

Maybe we should process JS and ESM, instead of CJS files. Plus jsx/tsx files.

Reply:
`info.path` is the output file path, cjs interop only works with cjs output